### ISSUE TITLE
Warning banner Tor Browser Androi

### ISSUE THREAD

**Issue by KwadroNaut** [freedomofpress/securedrop.org#687]
Tor browser for Android gives you the big red warning when that you're not anonymous when visiting pages like securedrop.org/directory. It'd be nice if it was a bit smarter. It currently includes a link to torproject.org, where one could donwload Tor browser for Android. Some people like recursivity, but maybe rewording to use the desktop version would be better. I don't know if removing the banner completely for Tor Browser Android (and maybe other versions?) is a good idea or not.

**Comment by harrislapiroff** [comment]
@redshiftzero, any chance you can tell me what you all ended up doing for Tor Browser detection on SecureDrop? I seem to remember this coming up there recently.

**Comment by eloquence** [comment]
FYI the current logic in the source interface can be found here: https://github.com/freedomofpress/securedrop/blob/develop/securedrop/static/js/source.js

**Comment by harrislapiroff** [comment]
ah, thanks @eloquence!

**Comment by harrislapiroff** [comment]
Marking this as ready for development. Let's update our detection to match what SecureDrop does. @rocodes, any caveats we need to know about the javascript here?

**Comment by chigby** [comment]
Are we sure we're not already using the same Tor Browser detection logic?

Compare SecureDrop

https://github.com/freedomofpress/securedrop/blob/ec2220c3c2b9120d029b616d3a07647b175bc6ab/securedrop/static/js/source.js#L71-L76

```js
function looksLikeTorBrowser() {
  return window.navigator.userAgent.match(TBB_UA_REGEX) &&
    new Date().getTimezoneOffset() == 0 &&
    window.screen.width == window.innerWidth &&
    window.screen.height == window.innerHeight;
}
```

With securedrop.org:

https://github.com/freedomofpress/securedrop.org/blob/c9f9f5f657e90c52828fa3ba57034a435630d4a6/client/tor/js/detect-tor.js#L5-L18

**Comment by SaptakS** [comment]
Shouldn't the regex (in both securedrop and securedrop.org) have Android mentioned in the tor browser as well? I see `(Windows NT 10\.0|X11; Linux x86_64|Macintosh; Intel Mac OS X 10\.14|Windows NT 10\.0; Win64; x64)` for tor browser in both places. While in securedrop codebase, there is `(Android; Mobile; rv:[0-9]{2}\.0\)` for an Orfox check which I have a feeling also handles the mobile Tor browser scenario (given both are based on firefox in android).
cc: @chigby

**Comment by chigby** [comment]
Oh, I completely overlooked that. :sweat_smile:  It looks like Orfox was a name of a former version of Tor Browser for Android? Then it seems like this additional check is what we must add.

**Comment by SaptakS** [comment]
>  It looks like Orfox was a name of a former version of Tor Browser for Android?

Orfox is a project which was maintained by GuardianProjects. They sunset their project after Tor released their own mobile browser (https://blog.torproject.org/orfox-paved-way-tor-browser-android). But at a javascript level, yeah, the userAgent check should be the same.
